packages/ai-ide-agents/src/browser/file-changeset-functions.ts

```diff
@@ -108,78 +106,134 @@ export class ReplaceContentInFileProvider implements ToolProvider {
-                                }
-                            },
-                            required: ['oldContent', 'newContent']
-                        },
-                        description: 'An array of replacement objects, each containing oldContent and newContent strings.'
-                    }
+    getToolMetadata(supportMutipleReplace: boolean = false): { description: string, parameters: ToolRequestParameters } {
+        const replacementProperties: ToolRequestParametersProperties = {
+            oldContent: {
+                type: 'string',
+                description: 'The exact content to be replaced. Must match exactly, including whitespace, comments, etc.'
+            },
+            newContent: {
+                type: 'string',
+                description: 'The new content to insert in place of matched old content.'
+            }
+        };
+
+        if (supportMutipleReplace) {
+            replacementProperties.multiple = {
+                type: 'boolean',
+                description: 'Set to true if multiple occurrences of the oldContent are expected to be replaced.'
+            };
+        }
+        const replacementParameters = {
+            type: 'object',
+            properties: {
+                path: {
+                    type: 'string',
+                    description: 'The path of the file where content will be replaced.'
                 },
-                required: ['path', 'replacements']
+                replacements: {
+                    type: 'array',
+                    items: {
+                        type: 'object',
+                        properties: replacementProperties,
+                        required: ['oldContent', 'newContent']
+                    },
+                    description: 'An array of replacement objects, each containing oldContent and newContent strings.'
+                }
             },
-            handler: async (args: string, ctx: MutableChatRequestModel): Promise<string> => {
-                try {
-                    const { path, replacements } = JSON.parse(args) as { path: string, replacements: Replacement[] };
-                    const fileUri = await this.workspaceFunctionScope.resolveRelativePath(path);
-                    const fileContent = (await this.fileService.read(fileUri)).value.toString();
+            required: ['path', 'replacements']
+        } as ToolRequestParameters;
 
-                    const { updatedContent, errors } = this.replacer.applyReplacements(fileContent, replacements);
+        const replacementSentence = supportMutipleReplace
+            ? 'By default, a single occurrence of each old content in the tuples is expected to be replaced. If the optional \'multiple\' flag is set to true, all occurrences will\
+             be replaced. In either case, if the number of occurrences in the file does not match the expectation the function will return an error, try a different approach then.'
```

Comment:
From 'try a different approach then,' this sentence is a run-on.

 ```suggestion
             be replaced. In either case, if the number of occurrences in the file does not match the expectation the function will return an error. In that case, try a different approach.'
```

Same below.